spec/scenarios/instant_charges_spec.rb

```diff
@@ -0,0 +1,524 @@
+      end
+
+      ### 18 february: Send an other event.
+      feb18 = DateTime.new(2023, 2, 18)
+
+      travel_to(feb18) do
+        expect do
+          create_event(
+            {
+              code: billable_metric.code,
+              transaction_id: SecureRandom.uuid,
+              external_customer_id: customer.external_id,
+              properties: { unique_id: 'id_2' },
+            },
+          )
+        end.to change { subscription.reload.fees.count }.from(2).to(3)
+
+        fee = subscription.fees.order(created_at: :desc).first
+
+        expect(fee.invoice_id).to be_nil
+        expect(fee.charge_id).to eq(charge.id)
+        expect(fee).to be_instant_charge
+        expect(fee.units).to eq(1)
+        expect(fee.events_count).to eq(1)
+        expect(fee.amount_cents).to eq(1200)
+      end
+    end
+  end
+
+  describe 'with sum_agg / package' do
+    let(:aggregation_type) { 'sum_agg' }
+    let(:field_name) { 'amount' }
+
+    it 'creates an instant fee' do
+      ### 24 january: Create subscription.
+      jan24 = DateTime.new(2023, 1, 24)
+
+      travel_to(jan24) do
+        create_subscription(
+          {
+            external_customer_id: customer.external_id,
+            external_id: customer.external_id,
+            plan_code: plan.code,
+          },
+        )
+      end
+
+      charge = create(
+        :package_charge,
+        :instant,
+        plan:,
+        billable_metric:,
+        properties: { amount: '100', free_units: 3, package_size: 2 },
+      )
+
+      subscription = customer.subscriptions.first
+
+      ### 15 february: Send an event.
+      feb15 = DateTime.new(2023, 2, 15)
+
+      travel_to(feb15) do
+        expect do
+          create_event(
+            {
+              code: billable_metric.code,
+              transaction_id: SecureRandom.uuid,
+              external_customer_id: customer.external_id,
+              properties: { amount: '3' },
+            },
+          )
+        end.to change { subscription.reload.fees.count }.from(0).to(1)
+
+        fee = subscription.fees.first
+
+        expect(fee.invoice_id).to be_nil
+        expect(fee.charge_id).to eq(charge.id)
+        expect(fee).to be_instant_charge
+        expect(fee.units).to eq(3)
+        expect(fee.events_count).to eq(1)
+        expect(fee.amount_cents).to eq(0) # free units
+      end
+
+      travel_to(DateTime.new(2023, 2, 17)) do
+        expect do
+          create_event(
+            {
+              code: billable_metric.code,
+              transaction_id: SecureRandom.uuid,
+              external_customer_id: customer.external_id,
+              properties: { amount: '1' },
+            },
+          )
+        end.to change { subscription.reload.fees.count }.from(1).to(2)
+
+        fee = subscription.fees.order(created_at: :desc).first
+        expect(fee.units).to eq(1)
+        expect(fee.events_count).to eq(1)
+        expect(fee.amount_cents).to eq(10000)
```

Comment:
Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them. (https://rubystyle.guide#underscores-in-numerics)